### DISCUSSION TITLE
Handling server side requests and cache in new NextJS 13

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2695] [Q&A] [answered]
I'm trying to follow the SDK's documentation regarding how to handle SSR for Next.JS, specifically this:

```
async function initPocketBase(req, res) {
  const pb = new PocketBase('http://127.0.0.1:8090');

  // load the store data from the request cookie string
  pb.authStore.loadFromCookie(req?.headers?.cookie || '');

  // send back the default 'pb_auth' cookie to the client with the latest store state
  pb.authStore.onChange(() => {
    res?.setHeader('set-cookie', pb.authStore.exportToCookie());
  });

  try {
      // get an up-to-date auth store state by verifying and refreshing the loaded auth model (if any)
      pb.authStore.isValid && await pb.collection('users').authRefresh();
  } catch (_) {
      // clear the auth store on failed refresh
      pb.authStore.clear();
  }

  return pb
}
```

However, I'm using the new app router, which deprecated `getServerSideProps`. I've been trying to port it myself, but with no success. The part that puzzles me is the `req,res` props, which I don't know how to use in this situation. I'll post a simple request I'm using, which gets indefinitely cached right now, even when new data is added.

```
export async function getUpdates() {
  const updates = await pb.collection("updates").getFullList();
  return updates;
}
```

One temporary "solution" I've found is to wrap the request in a `fetch` with `{revalidate}`, but it's far from ideal.

**Comment by ganigeorgiev** [comment, marked as answer]
I still haven't got the time to explore Next.js 13 in details but for the cache based on [their documentation](https://nextjs.org/docs/app/building-your-application/data-fetching/caching#per-request-caching), you can try configuring the [`fetchCache`](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#fetchcache) option, something like:

```js
export const fetchCache = 'default-no-store' // refer to the fetchCache docs for all available options
```

**Reply by g1sbi** [reply]
Thank you. I've found a similar temporary solution for now, which is to use at the top of a SSR page `export const revalidate = 60`. In this way I can choose the caching length based on each page.

**Reply by Dave-lab12** [reply]
Hi @g1sbi 
```javacript
export const fetchCache = 'default-no-store'
```
Works fine and cached all the data but revalidation is not working 
Does the code below revalidating every 60 seconds for you?
```javascript
export const revalidate = 60
```
